File: hw/chavez/u3/hw24AbstractFactory/Java/ClientApp/src/main/java/ec/edu/espe/clientapp/model/GUIFactory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.clientapp.model;

/**
 *
 * @author Odalys Chavez, CodeBreakers, @ESPE
 */
public abstract class GUIFactory {
    public static GUIFactory getFactory() {

        String osName = System.getProperty("os.name");

        if (osName.startsWith("Windows"))
            return new WinFactory();
        else
            return new LinuxFactory();
    }

    public abstract Button createButton();
    public abstract Menu createMenu();
    
}

File: hw/chavez/u3/hw24AbstractFactory/Python/model/gui_factory.py
from abc import ABC, abstractmethod
import platform

class GUIFactory(ABC):

    @staticmethod
    def get_factory():
        if platform.system() == "Windows":
            from model.win_factory import WinFactory
            return WinFactory()
        else:
            from model.linux_factory import LinuxFactory
            return LinuxFactory()

    @abstractmethod
    def create_button(self):
        pass

    @abstractmethod
    def create_menu(self):
        pass